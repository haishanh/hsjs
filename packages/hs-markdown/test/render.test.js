'use strict';

const markdown = require('..');

describe('model', () => {
  it('empty string', () => {
    const input = '';
    const ret = markdown.render(input);
    expect(ret).toMatchSnapshot();
  });

  it('all space string', () => {
    const input = '    ';
    const ret = markdown.render(input);
    expect(ret).toMatchSnapshot();
  });

  it('simple string', () => {
    const input = 'hello';
    const ret = markdown.render(input);
    expect(ret).toMatchSnapshot();
  });

  it('headings', () => {
    const input = `
# one

## two0

### three0

## two1

### three1
    `;
    const ret = markdown.render(input);
    expect(ret).toMatchSnapshot();
  });

  it('highlight', () => {
    const input = `
# one

` + '```js\nconsole.log("test")\n```\n' + `
## two
`;
    const ret = markdown.render(input);
    expect(ret).toMatchSnapshot();
  });
});
