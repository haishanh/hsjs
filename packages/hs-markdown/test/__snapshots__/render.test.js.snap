// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`model all space string 1`] = `
Object {
  "body": "",
  "toc": null,
}
`;

exports[`model empty string 1`] = `
Object {
  "body": "",
  "toc": null,
}
`;

exports[`model headings 1`] = `
Object {
  "body": "<h1 id=\\"one\\"><a href=\\"#one\\" aria-hidden class=\\"anchor\\">#</a>one</h1>
<h2 id=\\"two0\\"><a href=\\"#two0\\" aria-hidden class=\\"anchor\\">#</a>two0</h2>
<h3 id=\\"three0\\"><a href=\\"#three0\\" aria-hidden class=\\"anchor\\">#</a>three0</h3>
<h2 id=\\"two1\\"><a href=\\"#two1\\" aria-hidden class=\\"anchor\\">#</a>two1</h2>
<h3 id=\\"three1\\"><a href=\\"#three1\\" aria-hidden class=\\"anchor\\">#</a>three1</h3>",
  "toc": "<ul>
<li>
<p><a href=\\"#one\\">one</a></p>
<ul>
<li>
<p><a href=\\"#two0\\">two0</a></p>
<ul>
<li><a href=\\"#three0\\">three0</a></li>
</ul>
</li>
<li>
<p><a href=\\"#two1\\">two1</a></p>
<ul>
<li><a href=\\"#three1\\">three1</a></li>
</ul>
</li>
</ul>
</li>
</ul>",
}
`;

exports[`model highlight 1`] = `
Object {
  "body": "<h1 id=\\"one\\"><a href=\\"#one\\" aria-hidden class=\\"anchor\\">#</a>one</h1>
<div class=\\"highlight\\">
      <pre class=\\"language-js\\"><code>console<span class=\\"token punctuation\\">.</span><span class=\\"token function\\">log</span><span class=\\"token punctuation\\">(</span><span class=\\"token string\\">\\"test\\"</span><span class=\\"token punctuation\\">)</span></code></pre>
      </div>
<h2 id=\\"two\\"><a href=\\"#two\\" aria-hidden class=\\"anchor\\">#</a>two</h2>",
  "toc": "<ul>
<li>
<p><a href=\\"#one\\">one</a></p>
<ul>
<li><a href=\\"#two\\">two</a></li>
</ul>
</li>
</ul>",
}
`;

exports[`model simple string 1`] = `
Object {
  "body": "<p>hello</p>",
  "toc": null,
}
`;
